### DISCUSSION TITLE
[IOS] Moving to .Net 6 caused static assets from the IOS project to be missing at runtime

### DISCUSSION THREAD

**Discussion by arg-drummond** [unoplatform/uno#9836] [Q&A] [answered]
### Current behavior

Here is the directory structure of the IOS head that worked before with Xamarin:

```
Assets/
    Default-568h@2x.png
    SplashScreen@2x.png
    SpashScreen@3x.png
    Fonts/
         uno-fluentui-assets.tff
LaunchScreen.storyboard
```
Here is the extract from the csproj-file

```
	<ItemGroup>
		<InterfaceDefinition Include="LaunchScreen.storyboard" />
	</ItemGroup>
	
	<ItemGroup>
		<Compile Include="Main.cs" />
		<BundleResource Include="Assets\Fonts\uno-fluentui-assets.ttf" />
		<BundleResource Include="Assets\SplashScreen%402x.png" />
		<BundleResource Include="Assets\SplashScreen%403x.png" />
		<BundleResource Include="Assets\Default-568h%402x.png" />
	</ItemGroup>
```

Behaviour:
- All icons are missing and replaced by question marks.
- The images that should appear on the splashscreen are missing, while the splashscreen itself appears.

Images imported from the shared directory using absolute paths work as well.

### Expected behavior

The static imports work as previously.

### How to reproduce it (as minimally and precisely as possible)

Use above structure and target .NET6-ios.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. The structure of .NET 6 projects is different, and the assets need to be in a specific folder. See [this document](https://platform.uno/docs/articles/migrating-from-xamarin-to-net6.html) about migrating projects and let us know if there's something else missing.

**Reply by arg-drummond** [reply]
Thanks, renaming 'Assets' to 'Resources' worked indeed for the font. 
After still being unable to get the storyboard to function, I did a rebuild of the project from scratch changing nothing within the Uno project template except for adding my dependencies and removing conditionals. 
Building fails with this
![Screenshot 2022-09-15 142642](https://user-images.githubusercontent.com/112620248/190412794-8c29d773-583b-4383-beca-f22703006221.png)
error when using the unchanged folder structure. Placing the storyboard anywhere else causes the startup to be blank (not only missing images, but no storyboard at all).
There is definitely no duplicate and restarting the Mac didn't help either.

**Reply by arg-drummond** [reply]
PS: Could `<UseMaui>` be the issue? I

**Reply by jeromelaban** [reply]
Make sure to follow the patterns used by an app made by `dotnet new unoapp`. Also, `UseMaui` cannot be used with Uno apps. Where did you see that you need to use this property?

**Reply by arg-drummond** [reply]
To be able to use Maui.Essentials instead of Xamarin.Essentials. For some reason `<UsingMauiEssentials>True</UsingMauiEssentials>` alone does not include the required namespaces, I included Maui as a whole to get this fixed. (The Maui splashscreen even works as a replacement  but its splashscreen API is inherently buggy, I'd rather stick with Uno).
I am definitely following the recommended pattern. The template is unchanged except for the images which I replaced with mine and the following properties added to the project along my package/project dependencies:
```
<Nullable>enable</Nullable>
<UseMaui>True</UseMaui>
<UsingMauiEssentials>True</UsingMauiEssentials>
```

**Reply by jeromelaban** [reply]
Thanks. Which features are you using from Maui Essentials? 

It may not be possible to use Maui Essentials because of the ties to MAUI itself, so if you encounter issues while doing so, let us know as we may be able to adjust Uno.

**Reply by arg-drummond** [reply]
I am using 

`Microsoft.Maui.Devices` for display rotation,
`Microsoft.Maui.ApplicationModel` & `Microsoft.Maui.ApplicationModel.Communication` for the E-Mail API,
`Microsoft.Maui.Storage` for `SecureStorage `and `CacheDirectory` access.

I am not ruling out the issue being on MAUI's end as it should work on regular Xamarin.ios/-.android. [source](https://devblogs.microsoft.com/xamarin/xamarin-essentials-1-7-and-introducing-net-maui-essentials/).

Without support for `Maui.Essentials` in face of eventual deprecation of `Xamarin.Essentials`, do we have to settle with implementing everything natively again?

**Reply by jeromelaban** [reply]
Most of these APIs are already implemented by Uno. See Orientation with [SimpleOrientationSensor](https://github.com/unoplatform/uno/blob/340e1098870e36f5dffb96eac58479e8e2b13f8c/src/Uno.UWP/Devices/Sensors/SimpleOrientationSensor.cs#L6), Email with [EmailManager](https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UWP/ApplicationModel/Email/EmailManager.cs), Secure Storage with [PasswordVault](https://platform.uno/docs/articles/features/PasswordVault.html), for example.

**Reply by arg-drummond** [reply]
@jeromelaban 

Alright, it turns out the blog post is wrong or at least outdated on the usage of `<UsingMauiEssentials>True</UsingMauiEssentials>`. It sould have been `<UseMauiEssentials>True</UseMauiEssentials>` all along. Fixing this made `<UseMaui>True</UseMaui>` unnecessary and got the Uno splash screen to work again as expected. [This](https://github.com/dotnet/maui/issues/6102) also sounds a lot like there should no issues to be expected with Uno as long as only the Essentials are used.

While I can now build for debug and adhoc successfully, release builds fail with errors that would not occur with Xamarin before, such as
```
Failed to AOT compile SQLitePCLRaw.provider.sqlite3.dll, the AOT compiler exited with code 134
```
when using SQLitePCL

and 

```
ILStrip failed for C:/Git/MySolution/MyProject.Localization/bin/Release/netstandard2.1/en-Gb/MyProject.Localization.resources.dll: 
File '/Users/adrummond/Library/Caches/Xamarin/mtbs/builds/MyProject.iOS/aa2e5c207cf0/C:/Git/MySolution/MyProject.Localization/bin/Release/netstandard2.1/en-Gb/MyProject.Localization.resources.dll' not found.
```
when using resource files with a compile directive.

Do you happen to have any clue for me on where the causes might be? While it could be separate issues with .NET 6, I might as well just miss something in my project configuration specifically for release builds that the Uno templates do not account for.

**Reply by jeromelaban** [reply]
Thanks for the update on MauiEssentials. For SQLite, it's likely that this dependency needs to be updated to .NET 6 to work properly, but in all cases, it's using a part of the tooling that Uno does not manage. It's probably best to report this in the[ .NET iOS repo](https://github.com/xamarin/xamarin-macios), same goes for the other localization issue.

**Reply by arg-drummond** [reply]
@jeromelaban I managed to resolve these things, nothing of it seemed Uno-related, so I'm closing here.
Thanks again.

**Comment by queequac** [comment]
@jeromelaban I have also issues with assets on net6, but with Android.
I follow the approach for splashscreen as described [here](https://platform.uno/docs/articles/splash-screen.html).
It always worked fine with Xamarin, with net6 it behaves different depending on my settings.
Using Uno 4.4 it worked as long as I did not turn on LLVM. With Uno 4.5 it's not working at all. Here it always get compilation error `Resources\drawable\splash.xml(2): error APT2260: resource drawable/splashscreen (aka my.package:drawable/splashscreen) not found.`

Is that documentation outdated? Or what could be wrong here?

**Reply by jeromelaban** [reply]
4.5 should not have changed much with regards to the way android resources work. .NET 6 varies only with the location of the files (it should be under the `Android` folder) but otherwise should be the same.

**Reply by queequac** [reply]
Mhm, strange. As I said, works with 4.4, but breaks with 4.5. Also strange why LLVM prevents the resources to be generated correctly.
Maybe I can create a small repo tomorrow, so someone else can have a look at this.

**Reply by queequac** [reply]
Interesting, my file in the shared project (`./assets/splashscreen.scale-400.png`) results in a `splashscreen.png` in the corresponding drawable folder... with Uno 4.5 it is called `assets_splashscreen.png`, that's why the build did not work before updating the `splash.xml`. Strange thing...

**Reply by jeromelaban** [reply]
About Android, this is a breaking change we've included to avoid naming clashes between content folders. See the [4.5 release notes ](https://github.com/unoplatform/uno/releases/tag/4.5.9) for additional details (breaking changes section).

This probably means we need to adjust the docs if those rely on a specific name.
